crates/trigger-http2/src/outbound_http.rs

```diff
@@ -42,7 +43,7 @@ impl spin_factor_outbound_http::OutboundHttpInterceptor for OutboundHttpIntercep
             let server = self.server.clone();
             let resp_fut = async move {
                 match server
-                    .handle_trigger_route(req, route_match, CHAINED_CLIENT_ADDR)
+                    .handle_trigger_route(req, route_match, Scheme::HTTP, CHAINED_CLIENT_ADDR)
                     .await
                 {
                     Ok(resp) => Ok(Ok(IncomingResponse {
```

Comment:
I wonder if we should be setting the `SelfRequestOrigin` unconditionally in this interceptor. It seems like `self` requests ought to work in service-chained handlers? :thinking: